Add documentation for the deployment of Node.js Web Apps (#512)
pagopa/dx: apps/website/docs/pipelines/node-webapp-deploy.md

---
sidebar_position: 1

> sidebar_label: Deployment of a Node.js Web App

Reviewer: ```suggestion
sidebar_label: Deploying a Node.js Application to Azure App Service
```

---

# Deployment of a Node.js Web App

The
[Node.js WebApp Deployment](https://github.com/pagopa/dx/blob/main/.github/workflows/web_app_deploy.yaml)
is a template for workflows designed for building and deploying **Node.js** web applications (including Next.js) from a Turborepo monorepo to Azure App Services.

> 

Reviewer 2: Can you use Prettier to adjust the line breaks?

## How It Works

The workflow consists of two main jobs:

**Build Job (`build`)**:

> - Checks out the source code.

Reviewer: suggestion: I'd rather not describe the pipeline internals (which are subject to changes) but the required setup from the point of view of the user, ie:

```
The pipeline assumes the following setup for you repository:
- you have a monorepo setup with turborepo...
- the following npm tasks are implemented in package.json
```

arguments should be documented inside the workflow itself. Examples suffices here.

Author: I have found some requirements using copilot, and they are in the `Notes` section, but it'd be useful the contribution of someone who have a deep knowledge.

For the internals, I agree with you. However, I have noticed our documentation is not consistent. Some has details, some others don't.

Last consideration, examples are necessarily tied with arguments (at least for mandatory ones)

- Deploys the zip artifact using the `az webapp deploy` command:
  - If `use_staging_slot` is `true` (default), deploys to the `staging` slot.
  - If `use_staging_slot` is `false`, deploys directly to the `production` slot.
- If deployed to the `staging` slot, performs a slot swap using `az webapp deployment slot swap` to move the new version to `production`.

## Usage

To use this workflow, invoke it from your own GitHub Actions workflow file.

```yaml
name: Deploy My App Service

on:
  push:
    branches:
      - main

jobs:
  deploy_app:
    uses: pagopa/dx/.github/workflows/web_app_deploy.yaml@main
    name: Deploy Web App to Production
    with:
      workspace_name: 'my-web-app-workspace' # The name of your app's workspace in turbo.json
      environment: 'prod' # The target environment (e.g., dev, uat, prod)
      resource_group_name: 'my-resource-group'
      web_app_name: 'my-web-app'
      # use_staging_slot: true # Optional: default is true
      # use_private_agent: true # Optional: default is true
```

## Notes

- This workflow assumes your project is structured as a Turborepo monorepo.

> - The workspace being deployed must have a build script in its package.json.

Reviewer 2: ```suggestion
- The workspace being deployed must have a build script in its `package.json`.
```
